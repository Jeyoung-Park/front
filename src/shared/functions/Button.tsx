import * as React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import { TextInput } from 'react-native-gesture-handler';

interface ButtonProps {
	style: object;
	text: string;
	action: Function;
	opposite : boolean;
	disabled : boolean;
}

const Button = ({ style, text, action ,opposite,disabled }: ButtonProps) => {
	return (
		<TouchableOpacity
			style={[styles.container, style,{backgroundColor: !opposite ? "#60fac1" : "#FFFFFF",}]}
			onPress={()=>action()}
			disabled={disabled}
		>
			<Text style={{alignSelf:'center',color:opposite ? "#60fac1" : "#FFFFFF"}}>{text}</Text>
		</TouchableOpacity>

	);

};

export default Button;

const styles = StyleSheet.create({
	container: {
		marginHorizontal: 80,
		height: 50,
		borderRadius: 50,
		justifyContent:"center",
		borderWidth:1,
		borderColor : "#60fac1"
	},


});