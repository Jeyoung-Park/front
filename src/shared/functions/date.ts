// 날짜와 관련된 함수들입니다.

interface DateToYYMMDDInterface{
    date: Date | undefined;
    separator: string;
  }

// date 객체를 yymmdd 포맷으로 변경
// separator를 통해 원하는 형식으로 변경 가능 
export const dateToYYMMDD = ({
  date,
  separator = '',
}: DateToYYMMDDInterface) => {
  if (!date) return 'error';

  let year = date.getFullYear().toString().slice(2, 4);
  let month: number | string = date.getMonth() + 1;
  month = month >= 10 ? month : `0${month}`;
  let day: number | string = date.getDate();
  day = day >= 10 ? day : `0${day}`;

  return `${year}${separator}${month}${separator}${day}`;
};
