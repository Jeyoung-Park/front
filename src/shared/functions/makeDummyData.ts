import * as Factory from 'factory.ts';
import { RoomInterface } from '../../types';

interface CategoryInterface {
  categoryId: number;
  categoryName: string;
}

export const getDummyCategories = (iter: number) => {
  const categoryFactory = Factory.Sync.makeFactory<CategoryInterface>({
    categoryId: Factory.each(i => i),
    categoryName: '이름',
  });
  return categoryFactory.buildList(iter);
};

export const getDummyRooms=(iter:number)=>{
  const roomFactory=Factory.Sync.makeFactory<RoomInterface>({
    roomId:Factory.each(i=>i),
    platformId:Factory.each(i=>i),
    bossUserId:Factory.each(i=>i),
    roomName:'roomName',
    totalPrice:10000,
    maxUser:4,
    personalPrice:10000/4,
    recentPayment: new Date(),
  });
  return roomFactory.buildList(iter);
}