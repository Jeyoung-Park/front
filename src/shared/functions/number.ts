// 숫자와 관련된 함수들입니다.

export const numberWithComma = (number: number | undefined) => {
  if (!number) return 'error';
  let regexp = /\B(?=(\d{3})+(?!\d))/g;
  return number.toString().replace(regexp, ',');
};
