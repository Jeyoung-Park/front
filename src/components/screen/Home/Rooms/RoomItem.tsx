import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
} from 'react-native';
import {RoomInterface} from '../../../../types';
import Icon from 'react-native-vector-icons/AntDesign';
import {numberWithComma} from '../../../../shared/functions/number';
import {dateToYYMMDD} from '../../../../shared/functions/date';
const WINDOW_WIDTH = Dimensions.get('window').width;
interface RoomItemProps {
  navigation?: any;
  roomItem: RoomInterface;
}

const RoomItem = ({navigation, roomItem}: RoomItemProps) => {
  return (
    <TouchableOpacity style={styles.container} onPress={()=>{
      navigation.navigate('Room');
    }}>
      <Text style={styles.roomName}>{roomItem.roomName}</Text>
      <Text style={styles.platformName}>{roomItem.platformId}</Text>
      <View style={styles.iconsContainer}>
        {Array.from({length: 3}, (value, index) => {
          return (
            <Icon
              key={index}
              style={styles.icon}
              name="smile-circle"
              size={20}
              color="black"
            />
          );
        })}
      </View>
      <View style={styles.datePriceContainer}>
        <Text style={styles.datePriceText}>
          {dateToYYMMDD({date: roomItem.recentPayment, separator: '.'})}까지
        </Text>
        <Text style={styles.datePriceText}>
          <Text style={styles.priceHighlightText}>
            {numberWithComma(roomItem.personalPrice)}
          </Text>
          원
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default RoomItem;

const styles = StyleSheet.create({
  container: {
    //   height:100,
    width: (WINDOW_WIDTH - 32) / 2 - 16,

    borderColor: 'gray',
    borderRadius: 4,
    borderWidth: 1,

    padding: 16,
  },

  roomName: {
    fontSize: 16,
    // marginBottom:4,
  },
  platformName: {
    fontSize: 12,
    marginBottom: 16,
  },

  iconsContainer: {
    flexDirection: 'row',
    marginBottom: 8,
  },
  icon: {
    marginRight: 10,
  },

  datePriceContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  datePriceText: {
    fontSize: 12,
  },
  priceHighlightText: {
    fontSize: 14,
    fontWeight: 'bold',
  },
});
