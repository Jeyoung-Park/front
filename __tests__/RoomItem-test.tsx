import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import {getDummyRooms} from '../src/shared/functions/makeDummyData';
import RoomItem from '../src/components/screen/Home/Rooms/RoomItem';

const roomItemObject = {
  roomId: 1,
  platformId: 1,
  bossUserId: 1,
  roomName: 'roomName',
  totalPrice: 10000,
  personalPrice: 2500,
  maxUser: 4,
  recentPayment: new Date('2021-09-01'),
};

describe('RoomItem', () => {
  it('rendering test', () => {
    const rendered = render(<RoomItem roomItem={roomItemObject} />);

    expect(rendered).toMatchSnapshot();
  });
});
