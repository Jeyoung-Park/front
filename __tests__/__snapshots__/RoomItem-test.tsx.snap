// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomItem rendering test 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "gray",
      "borderRadius": 4,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 16,
      "width": 343,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    roomName
  </Text>
  <Text
    style={
      Object {
        "fontSize": 12,
        "marginBottom": 16,
      }
    }
  >
    1
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 8,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="icon-test"
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="icon-test"
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="icon-test"
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 20,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="icon-test"
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      21.09.01
      까지
    </Text>
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        2,500
      </Text>
      원
    </Text>
  </View>
</View>
`;
